tests/test_md.py

```diff
@@ -91,8 +81,13 @@ def integrate_once_through(
                 du_dx = dU_dx_fn(x_t, params, box, lamb)[0]
                 all_du_dxs.append(du_dx)
                 all_xs.append(x_t)
-                v_t = ca*v_t + np.expand_dims(cbs, axis=-1)*du_dx
-                x_t = x_t + v_t*dt
+
+                noise = np.random.randn(*v_t.shape)
+
+                v_mid = v_t + np.expand_dims(cbs, axis=-1) * du_dx
+
+                v_t = ca * v_mid + np.expand_dims(ccs, axis=-1) * noise
```

Comment:
note that I wasn't sure how to test providing an artificial set of rngs.... maybe we can call into cuRAND and give it the same seed? (XORWOW in particular)

Reply:
Ahh, good call.

I don't immediately see a convenient way to emulate cuRAND from numpy, to construct a test for identical trajectories. (For example, each thread gets its own seed https://docs.nvidia.com/cuda/curand/host-api-overview.html#order , so the emulator would have to be at least that detailed... If we find we need to do something like this elsewhere, perhaps the CUDA simulator in Numba may be handy? https://numba.pydata.org/numba-doc/dev/cuda/simulator.html )

Perhaps it would be good to add a short test that we can use MD to estimate known ensemble averages to within some tolerance, for a couple choices of timestep and collision rate.

For now I've reverted the test to not use random numbers in https://github.com/proteneer/timemachine/pull/459/commits/17cde2ab069cdeba1254ca945a5e86f76228c94e .

Reply:
https://pyculib.readthedocs.io/en/latest/curand.html

Might work if you use XORWOW, since its a fully deterministic algorithm (should be easy to try). But I'm also okay with just removing the noise for the test.

Reply:
Thanks!

Will leave as-is for now, but later I plan to revisit how to test for accurate sampling.